src/Elliptic/Systems/Xcts/Equations.cpp

```diff
@@ -0,0 +1,619 @@
+    const Scalar<DataVector>& conformal_factor,
+    const Scalar<DataVector>& lapse_times_conformal_factor,
+    const Scalar<DataVector>& conformal_factor_correction,
+    const Scalar<DataVector>&
+        lapse_times_conformal_factor_correction) noexcept {
+  get(*linearized_lapse_equation) +=
+      (pow<4>(get(conformal_factor)) *
+           get(lapse_times_conformal_factor_correction) +
+       4. * get(lapse_times_conformal_factor) * pow<3>(get(conformal_factor)) *
+           get(conformal_factor_correction)) *
+          ((5. / 12.) * square(get(extrinsic_curvature_trace)) +
+           2. * M_PI * (get(energy_density) + 2 * get(stress_trace))) +
+      5. * pow<4>(get(conformal_factor)) * get(conformal_factor_correction) *
+          (get(shift_dot_deriv_extrinsic_curvature_trace) -
+           get(dt_extrinsic_curvature_trace));
+}
+
+void add_distortion_hamiltonian_and_lapse_sources(
+    gsl::not_null<Scalar<DataVector>*> hamiltonian_constraint,
+    gsl::not_null<Scalar<DataVector>*> lapse_equation,
+    const Scalar<DataVector>&
+        longitudinal_shift_minus_dt_conformal_metric_square,
+    const Scalar<DataVector>& conformal_factor,
+    const Scalar<DataVector>& lapse_times_conformal_factor) noexcept {
+  get(*hamiltonian_constraint) -=
+      0.03125 * pow<7>(get(conformal_factor)) /
+      square(get(lapse_times_conformal_factor)) *
+      get(longitudinal_shift_minus_dt_conformal_metric_square);
+  get(*lapse_equation) +=
+      0.21875 * pow<6>(get(conformal_factor)) /
+      get(lapse_times_conformal_factor) *
+      get(longitudinal_shift_minus_dt_conformal_metric_square);
+}
+
+void add_linearized_distortion_hamiltonian_and_lapse_sources(
+    gsl::not_null<Scalar<DataVector>*> hamiltonian_constraint,
+    gsl::not_null<Scalar<DataVector>*> lapse_equation,
+    const Scalar<DataVector>&
+        longitudinal_shift_minus_dt_conformal_metric_square,
+    const Scalar<DataVector>& conformal_factor,
+    const Scalar<DataVector>& lapse_times_conformal_factor,
+    const Scalar<DataVector>& conformal_factor_correction,
+    const Scalar<DataVector>&
+        lapse_times_conformal_factor_correction) noexcept {
+  get(*hamiltonian_constraint) -=
+      0.03125 * pow<6>(get(conformal_factor)) *
+      (7. * get(conformal_factor_correction) -
+       2. * get(conformal_factor) / get(lapse_times_conformal_factor) *
+           get(lapse_times_conformal_factor_correction)) /
+      square(get(lapse_times_conformal_factor)) *
+      get(longitudinal_shift_minus_dt_conformal_metric_square);
+  get(*lapse_equation) +=
+      0.21875 * pow<5>(get(conformal_factor)) *
+      (6. * get(conformal_factor_correction) -
+       get(conformal_factor) / get(lapse_times_conformal_factor) *
+           get(lapse_times_conformal_factor_correction)) /
+      get(lapse_times_conformal_factor) *
+      get(longitudinal_shift_minus_dt_conformal_metric_square);
+}
+
+template <Geometry ConformalGeometry>
+void add_momentum_sources_impl(
```

Comment:
The formulas in this function (and some others) are complicated enough that references to concrete equation numbers of your favourite paper/book would be useful.  Please add something like 
"This function implements Eqs. (xx) of Baumgarte+Shapiro "Numerical Relativity"  at the relevant places.